### PR TITLE
_changes with _doc_ids filter handle deleted doc with no channels metadata

### LINKED ISSUE #1857: changes feed with doc_ids filter omits `deleted` properties
If I use the `doc_ids` filter with the `_changes` feed, to get only specific docIDs, the change objects in the response are missing the `deleted` property denoting a deleted revision.

Here's an example from one of the CBL/iOS test databases, where I first run the changes feed the normal way, but limiting it only to the revision of interest:
`GET :4984/attach_test/_changes?since=3&limit=1`

``` json
{
    "last_seq": "4", 
    "results": [
        {
            "changes": [
                {
                    "rev": "2-61de0ad4b61a3106195e9b21bcb69d0c"
                }
            ], 
            "deleted": true, 
            "id": "propertytest", 
            "seq": 4
        }
    ]
}
```

Now using the `doc_ids` filter:

`GET :4984/attach_test/_changes?filter=_doc_ids&doc_ids=["propertytest"]`

``` json
{
    "last_seq": 4, 
    "results": [
        {
            "changes": [
                {
                    "rev": "2-61de0ad4b61a3106195e9b21bcb69d0c"
                }
            ], 
            "id": "propertytest", 
            "seq": 4
        }
    ]
}
```

Note that `"deleted":true` is missing.

### PR DESCRIPTION
**Pull request by ajres**
For _changes with _doc_ids filter if doc has no channels metadata check _deleted property on revision body.

fixes #1857

---

This change is [Reviewable](https://reviewable.io/reviews/couchbase/sync_gateway/1861)

### CODE REVIEW

File: rest/changes_api.go [hunk 1]
```diff
@@ -332,17 +332,24 @@ func (h *handler) sendChangesForDocIds(userChannels base.Set, explicitDocIds []s
...
-		for channel, removal := range populatedDoc.Channels {
-			//Doc is tagged with channel or was removed at a sequence later that since sequence
-			if removal == nil || removal.Seq > options.Since.Seq {
-				//if the current user has access to this channel
-				if h.user.CanSeeChannel(channel) {
-					//If the doc has been removed
-					if removal != nil {
-						removedChannels = append(removedChannels,channel)
-						if removal.Deleted {
-							row.Deleted = true
```
Comment (adamcfraser) [inline rest/changes_api.go:345]: Does this handle the case where the document hasn't been removed from the channel, but has been deleted?

e.g. their sync function is still channeling the document based on oldDoc, but the document has been deleted.

Comment (ajres) [inline rest/changes_api.go:345]: Good question, I'll test manually first, but I think that one deserves a unit test as well.

Comment (ajres) [inline rest/changes_api.go:345]: There is a similar case where a user can delete a document by adding the "_deleted":true property to existing properties i.e. "Channels" in this case, I'll test that also.

File: rest/changes_api.go [hunk 2]
```diff
@@ -354,6 +357,25 @@ func (h *handler) sendChangesForDocIds(userChannels base.Set, explicitDocIds []s
 					}
 				}
 			}
+		} else if h.user.Channels().Contains(ch.UserStarChannel) {
+			userCanSeeDocChannel = true
+		}
```
Comment (adamcfraser) [inline rest/changes_api.go]: If the user has the star channel, we can skip the iteration over the doc's channels.  So maybe flip this if statement to 'if has star channel, else doc.channels > 0 as a performance optimization?

**Comment by adamcfraser** [comment]
Added one comment on the last set of changes.  Generally looks great - just a performance optimization question.

File: rest/changes_api.go (hunk 2 above)
Comment (ajres) [inline rest/changes_api.go]: Good call, I have pushed up the change.

**Comment by adamcfraser** [comment]
Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/couchbase/sync_gateway/1861)_

**Pull request merged**